server/src/main/java/io/druid/segment/indexing/granularity/UniformGranularitySpec.java

```diff
@@ -57,7 +57,9 @@ public UniformGranularitySpec(
     if (inputIntervals != null) {
       List<Interval> granularIntervals = Lists.newArrayList();
       for (Interval inputInterval : inputIntervals) {
-        Iterables.addAll(granularIntervals, this.segmentGranularity.getIterable(inputInterval));
+        for (Interval interval : segmentGranularity.getIterable(inputInterval)) {
```

Comment:
I am just wondering we can only trim the first and last intervals? I guess there are overlaps of `inputIntervals` and `interval` only in the first and the last one because intervals from `segmentGranularity.getIterable()` are continous and made from `inputIntervals`. Correct me if I am wrong.